backend/generalhw.pm

```diff
@@ -39,24 +39,33 @@ sub new {
 
 sub get_cmd {
     my ($self, $cmd) = @_;
+    my $args;
 
     my $dir = get_required_var('GENERAL_HW_CMD_DIR');
     if (!-d $dir) {
         die "GENERAL_HW_CMD_DIR is not pointing to a directory";
     }
+
+    # Add required arguments when needed
+    $args = get_var('GENERAL_HW_FLASH_ARGS') if ($cmd eq 'GENERAL_HW_FLASH_CMD' and get_var('GENERAL_HW_FLASH_ARGS'));
+
     $cmd = get_required_var($cmd);
     $cmd = "$dir/" . basename($cmd);
     if (!-x $cmd) {
         die "CMD $cmd is not an executable";
     }
+
+    $cmd .= " $args" if $args;
+
```

Comment:
can you just delete the blank lines, move the variable declaration onto the line of first use?

Or why not bring it all together:

```perl
my $args = get_var('GENERAL_HW_FLASH_ARGS') if ($cmd eq 'GENERAL_HW_FLASH_CMD' and get_var('GENERAL_HW_FLASH_ARGS'));
$cmd .= " $args" if $args;
```

Reply:
Fixed.

Reply:
Reverted, as `$cmd` is updated above and breaks comparison.